source/tyr/tyr/resources.py

```diff
@@ -82,7 +80,29 @@ def get(self):
 
 class Index(flask_restful.Resource):
     def get(self):
-        return {'jobs': {'href': url_for('jobs', _external=True)}}
+        response = {}
+        for endpoint in api.endpoints:
+            try:
+                response[endpoint] = {'href': url_for(endpoint, _external=True)}
+            except werkzeug.routing.BuildError:
+                logging.warning('Could not build url for endpoint \'{}\' '.format(endpoint))
+        return response
+
+
+class Status(flask_restful.Resource):
+    def get(self):
+        def check_db():
+            tyr_db = sqlalchemy.create_engine(current_app.config['SQLALCHEMY_DATABASE_URI'])
```

Comment:
This look a bit dangerous, you are creating a new connection pool to the database, that may or may not be closed.
You should take some inspiration from kirin's status: https://github.com/CanalTP/kirin/blob/master/kirin/resources.py#L47-L55

Reply:
Not really the same case, as Kirin queries a table defined in the model. The table 'alembic_version' is automatically generated.
But it does need to be changed in some way